### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by sduskis**
- Using ManagedChannel instead of ChannelPool in BigtableSession
- Adding a new BigtableOptions getDefaultOptions() method.

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 1]
```diff
@@ -397,18 +399,27 @@ protected ChannelPool createChannelPool(final ChannelPool.ChannelFactory channel
...
     managedChannels.add(channelPool);
     return channelPool;
   }
 
-  public static BigtableInstanceClient createInstanceClient()
+  /**
+   * Create a {@link BigtableInstanceClient}.  {@link BigtableSession} objects assume that
+   * {@link BigtableOptions} have a project and instance.  A {@link BigtableInstanceClient} does not
+   * require project id or instance id, so {@link BigtableOptions#getDefaultOptions()} may be used
+   * if
```
Comment (spollapally) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: documentation seems incomplete.

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/config/TestBigtableOptions.java [hunk 2]
```diff
@@ -84,12 +84,12 @@ public void testSerialization() throws IOException, ClassNotFoundException {
...
   @Test
   public void testNullStringsDontThrowExceptions() {
-     new BigtableOptions.Builder().build();
+    BigtableOptions.getDefaultOptions();
   }
 
   @Test
   public void testEmulator() {
-    Map<String, String> oldEnv = System.getenv();
+    Map<String, String> oldEnv = System.getenv();`
```
Comment (spollapally) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/config/TestBigtableOptions.java]: is the extra back tick here causing the build failure ?

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java [hunk 3]
```diff
@@ -33,4 +40,15 @@
   public void testBigtableConnectionExists() {
     Assert.assertEquals(BigtableConnection.class, BigtableConfiguration.getConnectionClass());
   }
+
+  @Test
+  public void testTable() throws IOException {
```
Comment (spollapally) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java]: This block seems duplicate, can it be moved to the common test module?

Comment (sduskis) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java]: This is definitely duplicated, but that's ok.  This is a unit test. The common test module is for integration tests only.

**Comment by spollapally** [comment]
LGTM. Looks like "Configuration" class import is missing or something to that effect is failing the build.

**Pull request merged**